### PR TITLE
Design Pipeline ownership

### PR DESCRIPTION
**Pull request by rakshith91**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/7060

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/tests/test_common_blob.py [hunk 1]
```diff
@@ -1833,6 +1834,17 @@ def test_set_blob_permission(self):
...
+    def test_transport_closed_only_once(self):
+        transport = RequestsTransport()
+        url = self._get_account_url()
+        credential = self._get_shared_key_credential()
+        blob_name = self._get_blob_reference()
+        with BlobServiceClient(url, credential=credential, transport=transport) as bsc:
+            assert transport.session is not None
+            with bsc.get_blob_client(self.container_name, blob_name) as bc:
+                assert transport.session is not None
+            assert transport.session is not None  # Right now it's None
```
Comment (KieranBrantnerMagee) [inline sdk/storage/azure-storage-blob/tests/test_common_blob.py]: "right now it's none" as we expect it to fail?  (In which case skip?)  Or some other meaning?

File: sdk/storage/azure-storage-queue/tests/test_queue.py [hunk 2]
```diff
@@ -963,6 +964,17 @@ def test_unicode_update_message_unicode_data(self, resource_group, location, sto
         self.assertIsInstance(message.expires_on, datetime)
         self.assertIsInstance(message.next_visible_on, datetime)
 
+    @ResourceGroupPreparer()
+    @StorageAccountPreparer(name_prefix='pyacrstorage')
+    def test_transport_closed_only_once(self, resource_group, location, storage_account, storage_account_key):
```
Comment (KieranBrantnerMagee) [inline sdk/storage/azure-storage-queue/tests/test_queue.py]: nit: you never consume location/resource_group.  Do you need them? (and in test_queue_async)

File: sdk/storage/azure-storage-blob/tests/test_common_blob.py (hunk 1 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-blob/tests/test_common_blob.py]: It meant it was none before this change..ill remove it.. :)

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py [hunk 3]
```diff
@@ -216,6 +216,26 @@ def _batch_send(
         except StorageErrorException as error:
             process_storage_error(error)
 
+class TransportWrapper(HttpTransport):
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: Could we please add a docstring to both this and the async wrapper class just to describe their purpose.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py [hunk 4]
```diff
@@ -216,6 +216,26 @@ def _batch_send(
...
+    def send(self, request, **kwargs):
+        return self._transport.send(request, **kwargs)
+
+    def open(self):
+        pass
+
+    def close(self):
+        pass
+
+    def __enter__(self, *args): # pylint: disable=arguments-differ
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: Enter does not take arg (that will remove the pylint disable ;))

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client_async.py [hunk 5]
```diff
@@ -124,3 +125,24 @@ def _create_pipeline(self, credential, **kwargs):
...
+    async def send(self, request, **kwargs):
+        return await self._transport.send(request, **kwargs)
+
+    async def open(self):
+        pass
+
+    async def close(self):
+        pass
+
+    async def __aenter__(self, *args): # pylint: disable=arguments-differ
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client_async.py]: Enter does not take arg (that will remove the pylint disable ;))

File: sdk/storage/azure-storage-blob/tests/test_common_blob_async.py [hunk 6]
```diff
@@ -2281,6 +2281,16 @@ def test_upload_to_url_file_with_credential(self):
...
+    async def test_transport_closed_only_once(self):
+        transport = AsyncioRequestsTransport()
+        url = self._get_account_url()
+        credential = self._get_shared_key_credential()
+        blob_name = self._get_blob_reference()
+        async with BlobServiceClient(url, credential=credential, transport=transport) as bsc:
+            assert transport.session is not None
+            async with bsc.get_blob_client(self.container_name, blob_name) as bc:
+                assert transport.session is not None
+            assert transport.session is not None  # Right now it's None
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/tests/test_common_blob_async.py]: Remove the comment here too ;)

**Pull request merged**